duckduckgo/iOS
# Update Bookmarks UI and extract Bookmarks database

Task/Issue URL: https://app.asana.com/0/72649045549333/1202533376181935/f
Tech Design URL:
CC:

**Description**:



**Steps to test this PR**:
1.  Test Migration.  Using a build from develop create bookmarks, favorites, folders.  Install new version and check they are migrated successfully (favorite merged with bookmark if same url, old database cleaned etc).
2. Test Clean install. Using a clean build from develop, ensure app starts and you can create bookmarks, favorites, folders, etc. 
3. Check that changes are preserved across app restarts.
4. Check dependency chain and how Main view is now instantiated. Validate app start/restart state and lock/auto clear.
5. Check add/edit bookmark / favorite from browsing menu and menu updates to show correct state on change.
6. Check bookmark manager view shows bookmarks and folder navigation works.
7. Check bookmark manager view allows editing of bookmarks and folders.
8. Use swipe actions in bookmark manager view to set as favorite and delete.
9. Search functionality in bookmark manager shows expected bookmarks and swipe to favorite/delete works. Search updates when DB state changes.
10. When using address bar bookmarks top 2 matching bookmarks appear in the suggestions, and what you see matches DB latest state.
11. Favorites can be re-ordered on the favorites tab.
12. Switching between bookmarks and favorites updates the UI correctly.
13. Add all bookmarks from from tab switcher.
14. Favicons are saved as expected for bookmarks when they are added
15. Favorites are show in overlay (not re-orderable or editable)
16. Favorites show on home screen and are re-orderable 
17. Favorites show edit / remove menu when long pressed
18. Bookmarks import & export works as expected.
 



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
